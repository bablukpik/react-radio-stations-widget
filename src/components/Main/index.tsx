import { useEffect } from 'react';
import { RootStateOrAny, useSelector, useDispatch } from 'react-redux';
import Header from '../Header';
import StationsList from '../StationsList';
import Footer from '../Footer';
import styles from './Main.module.css';
import { fetchStations } from '../../utils/helperFunctions';
import { actionTypes } from 'src/reducers/stationsReducer';

const Main = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    const url = 'https://fc15faa5-0e6b-4657-a9b0-37e876ee51d1.mock.pstmn.io/stations';
    (async () => {
      const fetchedStations = await fetchStations(url);
      if (fetchedStations 
        && Object.keys(fetchedStations).length > 0
      ) {
        dispatch({ type: actionTypes.setStations, payload: fetchedStations });
      }
    })();
  }, [dispatch]);
  const allStations = useSelector((store: RootStateOrAny) => store.allStations);
  return (
    <div className={styles.container}>
      <Header />
        {(!allStations || allStations?.stations.length === 0)
        ? <h1>Loading Stations...</h1>
        : <StationsList state={allStations} />}
      <Footer state={allStations} />
    </div>
  );
}

export default Main;
