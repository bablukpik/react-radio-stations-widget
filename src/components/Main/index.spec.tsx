import { FC } from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import store from 'src/store';

import Main from './index';

const Wrapper: FC = ({ children }) => (
    <Provider store={store}>
      {children}
    </Provider>
  );
  
describe('Main', () => {
  it('should render', () => {
    const { asFragment } = render(<Main />, { wrapper: Wrapper });
    expect(asFragment()).toMatchSnapshot();
  });
});
