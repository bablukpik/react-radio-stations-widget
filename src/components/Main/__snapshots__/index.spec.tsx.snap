// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <header
      class="widgetHeader"
    >
      <nav
        class="nav"
      >
        <button
          class="button"
          type="button"
        >
          <img
            alt="Back arrow icon"
            data-testid="backArrowIcon"
            src="back-arrow.png"
          />
        </button>
        <h1
          class="navHeading"
          data-testid="navHeading"
        >
          STATIONS
        </h1>
        <button
          class="button"
          type="button"
        >
          <img
            alt="Switch icon"
            data-testid="switchIcon"
            src="switch.png"
          />
        </button>
      </nav>
    </header>
    <div
      class="stationListWrapper"
    >
      <ul
        class="stationList"
        style="opacity: 0;"
      >
        <li
          class="listItem"
          id="66,6"
        >
          <span
            class="text"
          >
            Putin FM
          </span>
          <span
            class="text textBold"
          >
            66,6
          </span>
        </li>
        <li
          class="listItem"
          id="101,2"
        >
          <span
            class="text"
          >
            Dribbble FM
          </span>
          <span
            class="text textBold"
          >
            101,2
          </span>
        </li>
        <li
          class="listItem"
          id="99,4"
        >
          <span
            class="text"
          >
            Doge FM
          </span>
          <span
            class="text textBold"
          >
            99,4
          </span>
        </li>
        <li
          class="listItem"
          id="87,1"
        >
          <span
            class="text"
          >
            Ballads FM
          </span>
          <span
            class="text textBold"
          >
            87,1
          </span>
        </li>
        <li
          class="listItem"
          id="142,2"
        >
          <span
            class="text"
          >
            Maximum FM
          </span>
          <span
            class="text textBold"
          >
            142,2
          </span>
        </li>
      </ul>
    </div>
    <footer
      class="footer"
    />
  </div>
</DocumentFragment>
`;
